lib/pmf/pmf_main.c

```diff
@@ -74,7 +74,7 @@ int pmf_setup(void)
 		 * PMF service, if it is defined.
 		 */
 		if (pmf_svc_descs[ii].init != NULL) {
-			rc = pmf_svc_descs[ii].init();
+			rc = SPECULATION_SAFE_VALUE(pmf_svc_descs[ii].init)();
```

Comment:
I'm struggling to see what the threat is here. I guess it's possible that the memory just past the end of pmf_svc_descs[] array might be speculatively loaded into the cache, but what can an attacker do with that information? It cannot see that memory directly. This is in the cold boot path, before any attacker controllable code is executed, so it cannot induce different cache behaviour to observe timing differences and infer anything about that memory.

Reply:
As discussed offline, we have revised decisions around cold boot path.